### PR TITLE
dummy error gadgets

### PR DESCRIPTION
**Pull request by DreamWuGit**
first we prefer to implement dummy gadgets for various error types such that failed tx can pass the circuit at the moment. after this will do real implementation for each errors defined in #597 !

### CODE REVIEW

File: zkevm-circuits/src/evm_circuit/execution/error_oog_constant.rs [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+        assert_eq!(
+            run_test_circuits(
+                TestContext::<2, 1>::simple_ctx_with_bytecode(bytecode).unwrap(),
+                None
+            ),
+            Ok(())
+        );
+    }
+
+    fn test_stack_underflow(value: Word) {
```
Comment (lispc) [inline zkevm-circuits/src/evm_circuit/execution/error_oog_constant.rs:190]: what about moving these to push/pop gadget file?

File: eth-types/src/evm_types/opcode_ids.rs [hunk 2]
```diff
@@ -346,6 +346,19 @@ impl OpcodeId {
...
+        [
+            Self::CALL,
+            Self::DELEGATECALL,
+            Self::CALLCODE,
+            Self::STATICCALL,
+            Self::CREATE,
+            Self::CREATE2,
+        ]
+        .contains(self)
+    }
```
Comment (lispc) [inline eth-types/src/evm_types/opcode_ids.rs:361]: maybe later we need a method is_call and is_create. So another approach here is to add `is_call` AND `is_create`. And use `is_call() || is_create()` in caller side.

File: zkevm-circuits/src/evm_circuit/execution/error_oog_constant.rs (hunk 1 above)
Comment (DreamWuGit) [reply zkevm-circuits/src/evm_circuit/execution/error_oog_constant.rs:190]: My original plan is moving to real implementing stack overflow/underflow gadget file in the future. but OK to me for now

File: eth-types/src/evm_types/opcode_ids.rs (hunk 2 above)
Comment (DreamWuGit) [reply eth-types/src/evm_types/opcode_ids.rs:361]: agree !

File: zkevm-circuits/src/evm_circuit/execution/error_oog_constant.rs (hunk 1 above)
Comment (DreamWuGit) [reply zkevm-circuits/src/evm_circuit/execution/error_oog_constant.rs]: moved in [1f219ec](https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/647/commits/1f219ec84675ffe570ac3da396adb5140f0e0e0d)

File: eth-types/src/evm_types/opcode_ids.rs (hunk 2 above)
Comment (DreamWuGit) [reply eth-types/src/evm_types/opcode_ids.rs:358]: updated in [2ade20d](https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/647/commits/2ade20d8891a442aa89897ab594f2cd6779815d2) thanks !

**Review by pinkiebell — APPROVED** [review]
LGTM 👍

**Review by lispc — APPROVED** [review]

**Pull request merged**